ddf-rpc-api/src/main/java/com/connexta/ddf/platform/direct/PlatformMethods.java

```diff
@@ -0,0 +1,171 @@
+import ddf.action.Action;
+import ddf.action.ActionRegistry;
+import ddf.action.impl.ActionImpl;
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.opengis.filter.sort.SortBy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PlatformMethods implements MethodSet {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PlatformMethods.class);
+  private static final int ACTION_ID_QUERY_LIMIT = 500;
+
+  private final RpcFactory rpc = new RpcFactoryImpl();
+
+  private final Map<String, RpcMethod> methods;
+
+  private CatalogMethods catalogMethods;
+
+  private final FilterBuilder filterBuilder;
+
+  private CatalogFramework catalogFramework;
+
+  private ActionRegistry actionRegistry;
+
+  public PlatformMethods(
+      CatalogFramework catalogFramework,
+      ActionRegistry actionRegistry,
+      CatalogMethods catalogMethods,
+      FilterBuilder filterBuilder) {
+    this.catalogFramework = catalogFramework;
+    this.actionRegistry = actionRegistry;
+    this.catalogMethods = catalogMethods;
+    this.filterBuilder = filterBuilder;
+
+    Map<String, RpcMethod> builder = new HashMap<>();
+    builder.put(
+        "ddf.platform/getActions",
+        rpc.method(
+            this::getActionsByIds,
+            "Takes either a list of metacard ids or a query in the form of `ddf.catalog/query`"
+                + " and returns all actions available on the metacards. `params` takes: ONLY ONE OF `ids` "
+                + "(value: List(String - valid metacard ids), max ids:"
+                + ACTION_ID_QUERY_LIMIT
+                + ") or `query` (See parameters available on `ddf.catalog/query`).  "));
+    methods = builder;
+  }
+
+  @Override
+  public Map<String, RpcMethod> getMethods() {
+    return methods;
+  }
+
+  private Object getActionsByIds(Map<String, Object> params) {
+    Optional<List<String>> ids =
+        Optional.of(params)
+            .map(p -> p.get("ids"))
+            .filter(List.class::isInstance)
+            .map(idList -> (List<String>) idList);
+    Optional<Map<String, Object>> queryParams =
+        Optional.of(params)
+            .map(p -> p.get("query"))
+            .filter(Map.class::isInstance)
+            .map(paramMap -> (Map<String, Object>) paramMap);
+
+    final boolean exactlyOneParam = ids.isPresent() ^ queryParams.isPresent();
```

Comment:
Did you feel fancy when you did this

Reply:
I can't even take credit for it sadly:
https://github.com/connexta/ddf-jsonrpc/pull/33#discussion_r516840922